torchci/components/JobLinks.tsx

```diff
@@ -109,33 +113,132 @@ function DisableIssue({ job }: { job: JobData }) {
   }
 
   // At this point, we should show something. Search the existing disable issues
   // for a matching one.
   const issueTitle = `DISABLED ${testName}`;
+  const issueBody = formatDisableTestBody(job.failureCaptures!);
+
   const issues: IssueData[] = data.issues;
-  let issueLink;
-  let linkText;
-  let buttonStyle;
   const matchingIssues = issues.filter((issue) => issue.title === issueTitle);
 
+  return (
+    <DisableIssue
+      matchingIssues={matchingIssues}
+      issueTitle={issueTitle}
+      issueBody={issueBody}
+      isDisabledTest={true}
+    />
+  );
+}
+
+const jobNameRe = /^(.*) \(([^,]*),.*\)/;
+function transformJobName(jobName?: string) {
+  if (jobName == undefined) {
+    return null;
+  }
+
+  // We want to have the job name in the following format WORKFLOW / JOB (CONFIG)
+  const jobNameMatch = jobName.match(jobNameRe);
+  if (jobNameMatch !== null) {
+    return `${jobNameMatch[1]} (${jobNameMatch[2]})`;
+  }
+
+  return jobName;
+}
+
+function formatUnstableJobBody() {
+  return encodeURIComponent(
+    "> Please provide a brief reason on why you need to mark this job as unstable."
+  );
+}
+
+function UnstableJob({ job, label }: { job: JobData; label: string }) {
+  const swrKey = isFailure(job.conclusion) ? `/api/issue/${label}` : null;
+  const { data } = useSWR(swrKey, fetcher, {
+    // Set a 60s cache for the request, so that lots of tooltip hovers don't
+    // spam the backend. Since actually mutating the state (through filing a
+    // disable issue) is a pretty heavy operation, 60s of staleness is fine.
+    dedupingInterval: 60 * 1000,
+    refreshInterval: 60 * 1000, // refresh every minute
+  });
+
+  if (!isFailure(job.conclusion)) {
+    return null;
+  }
+
+  const jobName = transformJobName(job.name);
+  // Ignore invalid job name
+  if (jobName === null) {
+    return null;
+  }
+
+  // If we don't yet have any data, show a loading state.
+  if (data === undefined) {
+    return <span>{" | "} checking for disable jobs</span>;
+  }
+
+  // At this point, we should show something. Search the existing disable issues
+  // for a matching one.
+  const issueTitle = `UNSTABLE ${jobName}`;
+  const issueBody = formatUnstableJobBody();
+
+  const issues: IssueData[] = data.issues;
+  const matchingIssues = issues.filter((issue) =>
+    issueTitle.includes(issue.title)
+  );
+
+  return (
+    <DisableIssue
+      matchingIssues={matchingIssues}
+      issueTitle={issueTitle}
+      issueBody={issueBody}
+      isDisabledTest={false}
+    />
+  );
+}
+
+function DisableIssue({
+  matchingIssues,
+  issueTitle,
+  issueBody,
+  isDisabledTest,
+}: {
+  matchingIssues: IssueData[];
+  issueTitle: string;
+  issueBody: string;
+  isDisabledTest: boolean;
+}) {
+  let issueLink = `https://github.com/pytorch/pytorch/issues/new?title=${issueTitle}&body=${issueBody}`;
+  let linkText = isDisabledTest
+    ? "Disable test"
+    : issueTitle.includes("UNSTABLE")
+    ? "Unstable job"
+    : "Disable job";
```

Comment:
what is the case where disable job shows up?

very small nit: maybe it should be "mark job unstable" or something similar since unstable isn't really a verb

Reply:
I was thinking if I should add a link to disable job too, but finally decide against it for now.  If we want another button to disable job, it could be added later.